File: server/app/middleware/schema/aaaccount.py
from typing import Optional
from uuid import UUID

from pydantic import BaseModel

from app.middleware.schema.fi_data import FiDeposit, FiTermDeposit, FiRecurringDeposit, FiCreditCard, FiInsurance


class AAAccount(BaseModel):
    id: UUID
    phone: str
    name: Optional[str] = None
    email: Optional[str] = None
    dob: Optional[str] = None
    pan: Optional[str] = None
    fi_deposit: Optional[FiDeposit] = None
    fi_term: Optional[FiTermDeposit] = None
    fi_recurring: Optional[FiRecurringDeposit] = None
    fi_credit_card: Optional[FiCreditCard] = None
    fi_insurance: Optional[FiInsurance] = None

File: easyLedger/app/src/main/java/software/sauce/easyledger/network/model/AA/AAAccount.kt
package software.sauce.easyledger.network.model.AA

import com.google.gson.annotations.SerializedName

data class AAAccount(
    @SerializedName("id")
    var uuid: String,

    @SerializedName("phone")
    var phone: String,

    @SerializedName("name")
    var name: String?,

    @SerializedName("email")
    var email: String?,

    @SerializedName("dob")
    var dob: String?,

    @SerializedName("pan")
    var pan: String?,

    @SerializedName("fi_deposit")
    var FiDeposit: FIDeposit?,
)
